nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -37,7 +38,8 @@ let
 
     TRUSTED_PROXY_IP = cfg.trustedProxy;
   }
-  // (if cfg.smtp.authenticate then { SMTP_LOGIN  = cfg.smtp.user; } else {})
+  // (if (cfg.database.host != "/run/postgresql" && cfg.database.port != null) then { DB_PORT = toString cfg.database.port; } else {})
```

Comment:
This also can be solved better by `lib.optionalAttrs`

Reply:
Fixed!